src/core/args.c

```diff
@@ -376,6 +380,10 @@ static MVMObject * decont_arg(MVMThreadContext *tc, MVMObject *arg) {
                             MVM_exception_throw_adhoc(tc, "Expected native int argument, but got num"); \
                         case MVM_CALLSITE_ARG_STR: \
                             MVM_exception_throw_adhoc(tc, "Expected native int argument, but got str"); \
+                        case MVM_CALLSITE_ARG_INT: \
+                        case MVM_CALLSITE_ARG_UINT: \
+                            /* Ignore signedness mismatch to facilitate rebootstrapping */ \
```

Comment:
Should something different be done here after this and the NQP+Rakudo PRs are merged?

Reply:
Yes, once we're up and running again, MoarVM should become more picky about these mismatches.